pykafka/producer.py

```diff
@@ -241,3 +237,326 @@ def produce(self, messages):
         # only *some* messages when a leader changes. Therefore, we don't want
         # a random partition distribution changing that on the retry.
         self._produce(self._partition_messages(messages), 0)
+
+
+_QueueInfo = namedtuple('QueueInfo', ['lock', 'flush_ready',
+                                      'slot_available', 'queue',
+                                      'producer'])
+
+
+class QueueInfo(_QueueInfo):
+    """A packet of information for a producer queue
+
+    A QueueInfo object contains thread-synchronization primitives corresponding
+    to a single broker for this producer.
+
+    :ivar lock: The lock used to control access to shared resources for this
+        queue
+    :type lock: threading.Lock
+    :ivar flush_ready: A condition variable that indicates that the queue is
+        ready to be flushed via requests to the broker
+    :type flush_ready: threading.Condition
+    :ivar slot_available: A condition variable that indicates that there is
+        at least one position free in the queue for a new message
+    :type slot_available: threading.Condition
+    :ivar queue: The message queue for this broker. Contains messages that have
+        been supplied as arguments to `produce()` waiting to be sent to the
+        broker.
+    :type queue: collections.deque
+    :ivar messages_inflight: A counter indicating how many messages have been
+        enqueued for this broker and not yet sent in a request.
+    :type messages_inflight: int
+    :ivar producer: The producer to which this QueueInfo instance belongs
+    :type producer: :class:`pykafka.producer.AsyncProducer`
+    """
+    def __new__(cls,
+                lock,
+                flush_ready,
+                slot_available,
+                queue,
+                producer,
+                messages_inflight):
+        ins = super(QueueInfo, cls).__new__(
+            cls, lock, flush_ready, slot_available, queue, producer)
+        # store messages_inflight separately to allow updating
+        ins.messages_inflight = messages_inflight
```

Comment:
If you have a mutable value, then an object is preferable to a namedtuple. Otherwise, it's not really a tuple anymore. If you're worried about the size of the object, you can use `__slots__`, but realistically there aren't going to be many of these hanging around, so it's probably not necessary.

(disclaimer: i realize protocol.py has this otherwise. It's something I've been working on and changed my mind on recently.)